Web Analytics Maintenance Process [first draft]

anakravitz opened the pull request.

sbreitenother commented:
This is a rest PR from Ana. She will add the markdown and then I will review

sbreitenother commented:
@anakravitz I REALLY like the posts - if only more organizations would do this.

I made a bunch of tiny edits. The most substantial edit was to the intro - bullets do not work well in the summary on the home page (the bullets just get removed and everything is combined as a sentence). I tried to preserve the spirit of what you were trying to accomplish there.

As I said in my message to you, I would normally do this as a PR from my branch to yours, but I figured this is easier as you're still getting used to GH. Feel free to remove / roll back any of my changes. Also, if you want to make edits, make sure you fetch origin so that your local copy includes my changes.

Great work!!!

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+* Unexplained data discrepancies
+* Untagged marketing campaigns or site elements
+* Confusing and messy naming conventions
+* Etc…
```
I don't actually know what the 'et cetera' here would be! Can we either list out the remaining things, write a sentence summarizing the other problems, or strike this last bullet?

anakravitz replied:
OK, added several new bullets of common issues

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+* Confusing and messy naming conventions
+* Etc…
+
+## Why web analytics breaks
```
should this be `break` instead of `breaks`? (I'm honestly not sure which is grammatically correct, but this sounded a bit off to my ear.

anakravitz replied:
It sounds right to me, since it's singular. For example, we'd say "web analytics is the analysis of web data", not "web analytics are..." I'll switch it to "web analytics tracking" though in case it sounds weird to anyone else.

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+A web analytics implementation project often starts with quite a lot of fanfare and resources. There will usually be an audit and needs assessment process to determine what tracking needs to be implemented or fixed, an implementation project plan identifying task owners and dates, and earmarked hours from the development team for tasks like implementing tracking code and building a data layer. Al ...
+<!--more-->
+
+Common web analytics issues:
```
Insert newline here for markdown rendering.

anakravitz replied:
I added a '<br/>' to create a new line (let me know if there's a better way)

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+### Step 1: Audit the existing implementation
+If you already have a web analytics tool in place, methodically review the account setup and run a technical tracking review. 
+
+The audit should include the following elements, at least:
```
Insert newline here for markdown rendering

anakravitz replied:
Done

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+If you already have a web analytics tool in place, methodically review the account setup and run a technical tracking review. 
+
+The audit should include the following elements, at least:
+* All site tracking tags (standard pageview tags as well as any custom tags) to ensure they’re firing correctly
```
The second clause here "to ensure they’re firing correctly" reads a bit weird because it's not clear what the verb here is. I think it's "audit" or "review" but that doesn't work given the way the sentence introducing the  list is phrased.

anakravitz replied:
I changed it to "The following elements should be audited", please let me know if that works better

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+* Campaign tracking tags and marketing channel groups
+* Naming conventions for all custom events or variables
+* Admin setup in the analytics account, including custom features like goals, content grouping, filters, etc.
+* Overall account structure to ensure logical organization of all properties and views. For organizations with multiple sites, consider whether roll-up reporting should be enabled, and whether implementation should be standardized across sites for ease of reporting
```
Same issue here with lack of clarity about the verb that applies to "to ensure logical organization"

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+* Naming conventions for all custom events or variables
+* Admin setup in the analytics account, including custom features like goals, content grouping, filters, etc.
+* Overall account structure to ensure logical organization of all properties and views. For organizations with multiple sites, consider whether roll-up reporting should be enabled, and whether implementation should be standardized across sites for ease of reporting
+* A strategic review of the site to identify site and user elements that should be tracked, that are not yet. This will often involve discussions with different departments. 
```
The phrasing in the first sentence is a bit awkward. Maybe activate it to something like:
> A strategic review of the site to identify site and user elements that you want to start tracking.

anakravitz replied:
I changed it, I think it's a bit less clunky now

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+* Overall account structure to ensure logical organization of all properties and views. For organizations with multiple sites, consider whether roll-up reporting should be enabled, and whether implementation should be standardized across sites for ease of reporting
+* A strategic review of the site to identify site and user elements that should be tracked, that are not yet. This will often involve discussions with different departments. 
+
+For a new site, or a site without existing tracking, you do not need to fix anything yet, so will just need to plan what you would like to track.
```
"so will just need to plan" ->
> so you will just need to plan

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+|   | Goals  | Registration and purchase funnels | | |
+
+### Step 3: Implement
+Implement tags using your preferred method (ideally a tag management system). Set up goals, add any filters and data cleanup via Admin settings, QA thoroughly, and then launch to production.
```
I feel like "QA thoroughly" really masks a lot of hard work! If you have a good resource at hand for how to do this, maybe include a link?

anakravitz replied:
You're right, QAing is a big process, but there's not really a single resource for it since it depends so much on each specific tool and implementation. I added a couple links to plugins used for debugging 2 of the most popular web analytics tools.

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+Implement tags using your preferred method (ideally a tag management system). Set up goals, add any filters and data cleanup via Admin settings, QA thoroughly, and then launch to production.
+
+### Step 4: Build tracking into the core of new features
+Note that steps 1-3 are the standard steps of an implementation project, as described in the introduction. We can consider those to be the “launch” phase, while steps 4 and on are the “maintenance” phase designed to prevent post-launch degradation.
```
You could add a bit of hierarchy into the post to make this more clear.

## Part 1: launch
### Step 1 ...
### Step 2 ...
...
## Part 2: maintenance:
### Step 4...

anakravitz replied:
OK

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+### Step 4: Build tracking into the core of new features
+Note that steps 1-3 are the standard steps of an implementation project, as described in the introduction. We can consider those to be the “launch” phase, while steps 4 and on are the “maintenance” phase designed to prevent post-launch degradation.
+
+Building tracking into new features requires that analytics becomes as integral to new features as the basic functionality itself. For example, say the product team develops a new product recommendation component to be featured on the homepage. In an organization not optimized for web analytics, the new feature launches. Perhaps a day or a week later, the product team wants to understand the impa ...
```
I feel like "analytics" is a bit ambiguous here. Maybe "instrumentation" is more clear?

anakravitz replied:
I changed it to "web analytics implementation"

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,83 @@
+Running a full audit regularly is not practical due to the effort involved, but it is still important to keep on top of changes. To address this you can run regular mini audits every month or two to review and catch tracking and setup problems that might have otherwise been overlooked. Google lists [several tools](https://www.google.com/analytics/partners/search/apps?category=114) to automate the ...
+
+## Final thoughts 
+Ideally everyone should be looking at their slice of the business, not just designated Data Analysts. Besides the business benefits of democratizing and increasing data usage, having more eyes on the data will help quickly identify any broken tracking or other issues. And if there are no tracking issues, then people will put more faith in the data and use it more, which is the desired cycle. Foll ...
```
For this sentence: 
> And if there are no tracking issues, then people will put more faith in the data and use it more, which is the desired cycle.
It'd be great to get a back-link to my post on the data quality flywheel :)

mikekaminsky commented on content/post/web-analytics-process.md:
```diff
@@ -0,0 +1,90 @@
+---
+author: Ana Kravitz
+title: "Web Analytics Maintenance Process"
+publishDate: "2018-07-28"
```
@anakravitz we've got you scheduled for "2018-07-08" right now. If you're good with that you can change the publishdate and drop the `draft` tag :)

anakravitz commented:
@mikekaminsky cool, thanks for your edits/comments. I added a publish date and removed the draft tag. One thing is that it looks like the "relref" shortcode link to your post isn't working in final paragraph, can you pls check that when you get a chance?

mikekaminsky commented:
@anakravitz the link works for me. Can you tell me more about the issue you're seeing?

mikekaminsky commented:
Yep, I suspect it's a github markdown issue. It works when I run the Hugo
server, so we should be good to go :)

On Mon, Jun 18, 2018 at 8:25 PM, anakravitz <notifications@github.com>
wrote:

> @mikekaminsky <https://github.com/mikekaminsky> Not sure if I can add a
> screenshot here, but for me the markdown shows up as text rather than a
> link. Maybe that's because the relative link doesn't work in the Github
> preview and will work once it's on the actual site. If it's working for you
> then should be OK.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/locallyoptimistic/LocallyOptimistic/pull/84#issuecomment-398262633>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFJQYX0zu0KfkNEaUI6V8TPuDF8mMYYwks5t-G8fgaJpZM4UZOBt>
> .
>
